### ISSUE TITLE
BigQuery stateful ingestion fails at checkpoint construction

### ISSUE THREAD

**Issue by DSchmidtDev** [linkedin/datahub#4288]
**Describe the bug**
When I want to use stateful ingestion for bigquery sources it fails at constructing the state from a previously created checkpoint.
The first run works smoothly and creates the checkpoint, the second run tries to read it but fails. The message states that it cannot reconstruct the state because bigquery does not support platform instances. So writing and reading states may differ in constructing the models.
Therefore its not possible to use it with BigQuery.

**To Reproduce**
1. create a recipe using stateful ingestion pipeline
2. execute first run to create checkpoint
3. execute second run and see error in reading checkpoint

**Expected behavior**
Reconstructing the state from pipeline should work for bigquery sources and remove stale dataset entries.

**Error log**
```
File ".../python3.9/site-packages/datahub/ingestion/source/state/stateful_ingestion_base.py", line 175, in _get_last_checkpoint
    last_checkpoint = Checkpoint.create_from_checkpoint_aspect(
File ".../python3.9/site-packages/datahub/ingestion/source/state/checkpoint.py", line 99, in create_from_checkpoint_aspect
    logger.warning(
Message: "Failed to construct checkpoint's config from checkpoint aspect."
Arguments: (ConfigurationError('BigQuery project ids are globally unique. You do not need to specify a platform instance.'),)
```

**Desktop:**
 - Version `acry-datahub[bigquery]==0.8.27.1`

**Additional context**
https://datahubspace.slack.com/archives/C029A3M079U/p1645805026218839

**Comment by DSchmidtDev** [comment]
https://github.com/linkedin/datahub/pull/4295

**Comment by DSchmidtDev** [comment]
Fixed by #4295
